source/simulator/parameters.cc

```diff
@@ -949,6 +985,49 @@ namespace aspect
     }
     prm.leave_subsection ();
 
+
+    prm.enter_subsection ("Formulation");
+    {
+      // The following options need a lot of error checking, however, most of
+      // the information is not available at this point. Therefore the error checking is done
+      // in Simulator<dim>::check_consistency_of_formulation() after the initialization of
+      // material models, heating plugins, and adiabatic conditions.
+      combined_formulation = CombinedFormulation::parse(prm.get("Formulation"));
+      if (combined_formulation == CombinedFormulation::isothermal_compression)
+        {
+          formulation_mass_conservation = FormulationMassConservation::isothermal_compression;
+          formulation_temperature_equation = FormulationTemperatureEquation::real_density;
+        }
+      else if (combined_formulation == CombinedFormulation::BA)
+        {
+          formulation_mass_conservation = FormulationMassConservation::incompressible;
+          formulation_temperature_equation = FormulationTemperatureEquation::reference_density_profile;
+
+          // Assert AdiabaticConditions = ConstantTemperature, density=1
+          // Assert shear/adiabatic heating plugins are off
+          // is_compressible= false
```

Comment:
If you want to keep these, maybe prefix with a sentence that states that these checks happen in the `check_consistency...` function mentioned above.

Or add a sentence above where you introduce this function to say that each of the blocks below has a group of conditions that need to hold in each case, and that are actually checked in that function.